### ISSUE TITLE
SV-DKL Approach / dynamic change of label amount

### ISSUE THREAD

**Issue by cherepanovic** [cornellius-gp/gpytorch#1075]
what is the best way to extend the model for an additional class after having trained a model. 

In the case of CIFAR, for example, I could train a model for five classes and would add next classes after having trained the model for 5 classes... Are there some concepts which could allow it?

Thanks! 

(accidently, I took an wrong label)

**Comment by gpleiss** [comment]
This isn't something that we've ever tried to do (nor is it something that is listed in the paper). There are many possible ways you could go about tackling this though.

Can you post a code example of your model? If you are using the variant of SoftmaxLikelihood with `mixing_parameters`, one thing you could do is expand the `mixing_parameters` to account for the new class.

In more detail, the mixing_parameters are `f x c`, where `f` is the number of features and `c` is the number of classes. You could extend the parameters to be `f x (c + 1)`. Then - you could run some more iterations of gradient descent through the ELBO, but only pass in the `likelihood` parameters and not the deep net + gp parameters.

Again, there are many other ways you could go about tackling this problem (as there are many possible ways to extend any ML classification model to an additional class).
